src/presign/participant.rs

```diff
@@ -947,55 +952,62 @@ pub(crate) struct PresignKeyShareAndInfo {
 }
 
 impl PresignKeyShareAndInfo {
-    /// Corresponds to pre-signing round 1 for party i
-    ///
-    /// Produces local shares k and gamma, along with their encrypted
-    /// components K = enc(k) and G = enc(gamma).
+    fn new(id: ParticipantIdentifier, input: &Input) -> Result<Self> {
+        Ok(Self {
+            aux_info_private: input.auxinfo_private.clone(),
+            aux_info_public: input.find_auxinfo_public(id)?.clone(),
+            keyshare_private: input.keyshare_private.clone(),
+            keyshare_public: input.find_keyshare_public(id)?.clone(),
+        })
+    }
+
+    /// Round one of the presign protocol.
     ///
-    /// The public_keys parameter corresponds to a KeygenPublic for
-    /// each of the other parties.
+    /// This round produces two local shares `k` and `ɣ` along with their
+    /// ciphertexts encrypted using this participant's Paillier encryption key,
+    /// alongside a zero knowledge proof that the ciphertext associated with `k`
+    /// was constructed correctly. We utilize the `other_auxinfos` argument to
+    /// construct this proof, as it has to be tailored to the particular
+    /// receiving participant.
     #[cfg_attr(feature = "flame_it", flame("presign"))]
-    pub(crate) fn round_one<R: RngCore + CryptoRng>(
+    fn round_one<R: RngCore + CryptoRng>(
```

Comment:
question: do you think there is a good reason to keep this as a function on a separate type (`PresignKeyShareAndInfo` / later `Input`) rather than integrating it inline into the `PresignParticipant`?

Reply:
I don't, but I'd like to save that decision for #351, as during that refactor I think we'll have a better sense of whether this separation is useful or not.